x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/searchablesnapshots/SearchableSnapshotFeatureSetUsage.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.core.searchablesnapshots;
+
+import org.elasticsearch.Version;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.xcontent.ToXContent;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.xpack.core.XPackFeatureSet;
+import org.elasticsearch.xpack.core.XPackField;
+
+import java.io.IOException;
+import java.util.Objects;
+
+public class SearchableSnapshotFeatureSetUsage extends XPackFeatureSet.Usage {
+
+    private final int numberOfSearchableSnapshotIndices;
+
+    public SearchableSnapshotFeatureSetUsage(StreamInput input) throws IOException {
+        super(input);
+        numberOfSearchableSnapshotIndices = input.readVInt();
+    }
+
+    @Override
+    public Version getMinimalSupportedVersion() {
+        return Version.V_8_0_0;
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        super.writeTo(out);
+        out.writeVInt(numberOfSearchableSnapshotIndices);
+    }
+
+    public SearchableSnapshotFeatureSetUsage(boolean available, boolean enabled,
+                                             int numberOfSearchableSnapshotIndices) {
+        super(XPackField.SEARCHABLE_SNAPSHOTS, available, enabled);
```

Comment:
Since it can't be disabled, we may as well inline this argument?

```suggestion
        super(XPackField.SEARCHABLE_SNAPSHOTS, available, true);
```

Reply:
sure, done in af2b703